src/sentry/integrations/slack/integration.py

```diff
@@ -120,15 +124,11 @@ def get_pipeline_views(self):
     def get_team_info(self, access_token):
         payload = {"token": access_token}
 
-        session = http.build_session()
-        resp = session.get("https://slack.com/api/team.info", params=payload)
-        resp.raise_for_status()
-        status_code = resp.status_code
-        resp = resp.json()
-        # TODO: track_response_code won't hit if we have an error status code
-        track_response_code(status_code, resp.get("ok"))
-
-        # TODO: check for resp["ok"]
+        client = SlackClient()
+        try:
+            resp = client.get("/team.info", params=payload)
+        except ApiError as e:
+            logger.error("slack.team-info.response-error", extra={"error": six.text_type(e)})
 
         return resp["team"]
```

Comment:
@MeredithAnya This line is going to error out if we get an `ApiError` as `resp` won't be defined. We definitely should raise the error above after catching it.